source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+
+```js
+cy.route2('/users/**', (req) => {
+  req.send({
+    statusCode: 200,
+    body: JSON.stringify({
+      profile: {
+        firstName: 'Tony',
+        lastName: 'Jarvis'
+      }
+    })
+  })
+})
+```
+
+In addition, because you have access to the request, this means that you can modify the request as well:
+
+```js
+cy.route2('/users/**', (req) => {
+  req.headers = { ...req.headers, accept: 'application/json' }
+  req.body = { ...req.body, note: 'Custom note' }
+})
+```
+
+**{% fa fa-angle-right %} method** ***(String)***
+
+Match the route to a specific HTTP method (`GET`, `POST`, `PUT`, etc).
+
+{% note bolt %}
+By default, Cypress will match `ALL` requests by default.
+{% endnote %}
+
+## Yields {% helper_icon yields %}
+
+{% yields null_alias cy.route2 %}
+
+# Examples
+
+## Without Stubbing
+
+If you do not pass a `response` to a route, Cypress will pass the request through without stubbing it. We can still wait for the request to resolve later.
+
+### Wait on `GET` request matching `url`
+
+```javascript
+cy.route2('**/users').as('getUsers')
+cy.visit('/users')
+cy.wait('@getUsers')
+```
+
+### Wait on matching `method` and `url`
+
+```javascript
+cy.route2('POST', '**/users').as('postUser')
+cy.visit('/users')
+cy.get('#first-name').type('Julius{enter}')
+cy.wait('@postUser')
+```
+
+### Wait on `url` matching glob
+
+Under the hood Cypress uses {% url 'minimatch' https://github.com/isaacs/minimatch %} to match glob patterns of `url`.
+
+This means you can take advantage of `*` and `**` glob support. This makes it *much* easier to route against dynamic segments without having to build up a complex `RegExp`.
+
+We expose {% url `Cypress.minimatch` minimatch %} as a function that you can use in your console to test routes.
+
+### Match route against any UserId
+
+```javascript
+cy.route2('**/users/*/comments')
+
+// https://localhost:7777/users/123/comments     <-- matches
+// https://localhost:7777/users/123/comments/465 <-- does not match
+```
+
+### Use glob to match all segments
+
+```javascript
+cy.route2('**/posts/**')
+
+// https://localhost:7777/posts/1            <-- matches
+// https://localhost:7777/posts/foo/bar/baz  <-- matches
+// https://localhost:7777/posts/quuz?a=b&1=2 <-- matches
+```
+
+### Modify request
+
+Before a request gets passed through, we can modify the request before it is sent out through a function. The function exposes a `req` object which contains the request.
+
+You can also (optionally) add a simulated {% urlHash 'delay' Simulate-delay %} or {% urlHash 'throttle' Simulate-throttle %} on the request to be sent out.
+
+```javascript
+cy.route2('/users', (req) => {
+  // modify the headers or body
+  req.headers = {...req.headers, accept: 'application/json'}
+  req.body = {...req.body, foo: 'bar'}
+})
+```
```

Comment:
see suggestions from where this example was used earlier

Reply:
see resolution to https://github.com/cypress-io/cypress-documentation/pull/2854/files/45ae8f49408128281ba5864dfa59354521b1d587#diff-6066033e28279d50ba94ae43b9b5b5a6